x-pack/plugins/secops/server/lib/network_top_n_flow/elasticsearch_adapter.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { getOr } from 'lodash/fp';
+
+import {
+  NetworkTopNFlowData,
+  NetworkTopNFlowEdges,
+  NetworkTopNFlowType,
+} from '../../graphql/types';
+import { FrameworkAdapter, FrameworkRequest } from '../framework';
+import { TermAggregation } from '../types';
+import { NetworkTopNFlowRequestOptions } from './index';
+import { buildQuery } from './query.dsl';
+import { NetworkTopNFlowAdapter, NetworkTopNFlowBuckets } from './types';
+
+export class ElasticsearchNetworkTopNFlowAdapter implements NetworkTopNFlowAdapter {
+  constructor(private readonly framework: FrameworkAdapter) {}
+
+  public async getNetworkTopNFlow(
+    request: FrameworkRequest,
+    options: NetworkTopNFlowRequestOptions
+  ): Promise<NetworkTopNFlowData> {
+    const response = await this.framework.callWithRequest<NetworkTopNFlowData, TermAggregation>(
+      request,
+      'search',
+      buildQuery(options)
+    );
+    const { cursor, limit } = options.pagination;
+    const totalCount = getOr(0, 'aggregations.network_top_n_flow_count.value', response);
+    const buckets = getOr([], 'aggregations.network_top_n_flow.buckets', response);
+    const networkTopNFlowItemAttr =
+      options.networkTopNFlowType === NetworkTopNFlowType.source ? 'source' : 'destination';
+    const networkTopNFlowEdges: NetworkTopNFlowEdges[] = buckets.map(
+      (bucket: NetworkTopNFlowBuckets) => ({
+        node:
+          bucket.domain.buckets.length > 0
+            ? {
+                _id: bucket.key,
+                [networkTopNFlowItemAttr]: {
+                  domain:
+                    bucket.domain.buckets[0].key === '__missing__'
```

Comment:
Couldn't you also just use `''` as your missing value so that if you have mix of missing domains and empty string domains it will aggregate them together and you don't have to check for a sentinel value here of `__missing__`? I think the intent of `missing` is to have you group missing into the same bucket as something you could already be aggregating on and let Elastic Search manage it within aggregates rather than within user land code?

Otherwise, I would almost suggest to be absolutely sure you don't clash with a bizarre use case of a customer having this `__missing__` value to use a constant on the backend and give it a GUID.

Something like:

```
const MISSING_SENTIENL = `__missing__be68598f-deb1-4fbc-8578-305d32871edf`
```
and then just check against that value. Better safe than sorry.

Reply:
I can see what you mean, but that's a reserved word from elastic search. So I feel that we should be fine.